### ISSUE TITLE
Create Role and  Management

### ISSUE THREAD

**Issue by kapendra-sentieo** [mongodb/mongodb-kubernetes-operator#1162]
I have a scenario where I need to create and assign a mongo role with a needed DB. Below is the config but the user is not reflected in mongo

```

---
# Source: as-mongodb/templates/deployment.yaml
apiVersion: mongodbcommunity.mongodb.com/v1
kind: MongoDBCommunity
metadata:
  namespace: mongodb
  name: mongodb
spec:
  members: 3
  type: ReplicaSet
  version: 4.4.17
  security:
    authentication:
      modes: ["SCRAM"]
    roles: # custom roles are defined here
      - role: read-cust
        db: test-db
        privileges:
          - resource:
              db: test-db
              collection: "" # an empty string indicates any collection
            actions: ["update","insert","remove"]
        roles: []
  users:
    - name: mongodb-admin
      db: admin
      passwordSecretRef: # a reference to the secret that will be used to generate the user's password
        name: mongodb-admin-password
      roles:
        - name: clusterAdmin
          db: admin
        - name: userAdminAnyDatabase
          db: admin
      scramCredentialsSecretName: my-scram
    - name: test-db-user
      db: test-db
      passwordSecretRef: # a reference to the secret that will be used to generate the user's password
        name: mongodb-admin-password
      roles:
        - name: read-cust
          db: test-db
  additionalMongodConfig:
    storage.wiredTiger.engineConfig.journalCompressor: zlib
```

**Comment by kapendra-sentieo** [comment]
any update

**Comment by kapendra-sentieo** [comment]
any update?

**Comment by mircea-cosbuc** [comment]
Hi @kapendra-sentieo ,
It looks like you are correctly creating the custom role and user in the `test-db` database, how are you testing if the user is reflected in the database?

**Comment by kapendra-sentieo** [comment]
I tested it by logging cluster with the admin user and then using db `test-db`.  then used the command `db.getRoles()`

I got [] in response. let me upload a screenshot in a while

**Comment by mircea-cosbuc** [comment]
@kapendra-sentieo Replicating your example, I created the resource like this:
```shell
cat <<EOF | kubectl -n mongodb apply -f -
---
# Source: as-mongodb/templates/deployment.yaml
apiVersion: mongodbcommunity.mongodb.com/v1
kind: MongoDBCommunity
metadata:
  namespace: mongodb
  name: mongodb
spec:
  members: 3
  type: ReplicaSet
  version: 4.4.17
  security:
    authentication:
      modes: ["SCRAM"]
    roles: # custom roles are defined here
      - role: read-cust
        db: test-db
        privileges:
          - resource:
              db: test-db
              collection: "" # an empty string indicates any collection
            actions: ["update","insert","remove"]
        roles: []
  users:
    - name: mongodb-admin
      db: admin
      passwordSecretRef: # a reference to the secret that will be used to generate the user's password
        name: mongodb-admin-password
      roles:
        - name: clusterAdmin
          db: admin
        - name: userAdminAnyDatabase
          db: admin
      scramCredentialsSecretName: my-scram
    - name: test-db-user
      db: test-db
      passwordSecretRef: # a reference to the secret that will be used to generate the user's password
        name: mongodb-admin-password
      roles:
        - name: read-cust
          db: test-db
      scramCredentialsSecretName: read-cust-scram
  additionalMongodConfig:
    storage.wiredTiger.engineConfig.journalCompressor: zlib
EOF
```
then ran a `mongosh` container to connect to the replicaset:
```
kubectl -n mongodb run --rm -it mongosh --image=mongo -- bash
If you don't see a command prompt, try pressing enter.
root@mongosh:/# mongosh "mongodb://mongodb-admin:<your-admin-password>@mongodb-0.mongodb-svc.mongodb.svc.cluster.local:27017,mongodb-1.mongodb-svc.mongodb.svc.cluster.local:27017,mongodb-2.mongodb-svc.mongodb.svc.cluster.local:27017/admin?replicaSet=mongodb&ssl=false"
...
mongodb [primary] admin> use test-db
switched to db test-db
mongodb [primary] test-db> db.getRoles()
{
  roles: [
    {
      role: 'read-cust',
      db: 'test-db',
      isBuiltin: false,
      roles: [],
      inheritedRoles: []
    }
  ],
  ok: 1,
  '$clusterTime': {
    clusterTime: Timestamp({ t: 1671554413, i: 10 }),
    signature: {
      hash: Binary(Buffer.from("64409e476d36a66ac62397ef51b2318f65586bb8", "hex"), 0),
      keyId: Long("7179270596721639428")
    }
  },
  operationTime: Timestamp({ t: 1671554413, i: 10 })
}
```
The only thing I can see missing from your example is the `scramCredentialsSecretName` for your custom user. This is a required field, but the resource would be rejected by the validation webhook unless you disabled it. This might explain why your user was not created.

**Comment by kapendra-sentieo** [comment]
let me try this.

**Comment by kapendra-sentieo** [comment]
thanks its working now.

**Comment by dan-mckean** [comment]
Hi, I'm Dan and I'm the Product Manager for MongoDB's support of Kubernetes.

I'm doing some work right now to try and identify how the Community Operator is being used. The Community Operator is something I inherited when I started at MongoDB, but it doesn't get as much attention from us as we'd like and we're trying to understand how it's used in order to establish its future. It will help us prioritize future issues and PRs raised by the community 🙂

Here's a super short survey (it's much easier for us to review all the feedback that way!): [https://docs.google.com/forms/d/e/1FAIpQLSfwrwyxBSlUyJ6AmC-eYlgW_3JEdfA48SB2i5--_WpiynMW2w/viewform?usp=sf_link](https://docs.google.com/forms/d/e/1FAIpQLSfwrwyxBSlUyJ6AmC-eYlgW_3JEdfA48SB2i5--_WpiynMW2w/viewform?usp=sf_link)

 If you'd rather email me instead: [dan.mckean@mongodb.com](mailto:dan.mckean@mongodb.com?subject=MongoDB%20Community%20Operator%20feedback)

Thank you in advance!
Dan
